python/ray/data/block.py

```diff
@@ -512,21 +512,47 @@ def block_type(self) -> BlockType:
         raise NotImplementedError
 
 
-def _get_block_boundaries(columns: list[np.ndarray]) -> np.ndarray:
-    """Compute boundaries of the groups within a block, which is represented
-    by a list of 1D numpy arrays for each column. In each column,
-    NaNs/None are considered to be the same group.
+def _get_group_boundaries_sorted(
+    block: Block, key: Union[str, List[str]]
```

Comment:
> key: Union[str, List[str]]

not a new issue, but I think we should stop handling key-or-keys everywhere. Ideally only the user-facing API needs to deals with single or multiple keys. 
All internal function should only handle lists to simplify code

Reply:
On the same page. Let me fix that